src/app/static/src/app/components/ChoroplethFilters.vue

```diff
@@ -0,0 +1,161 @@
+            selectedDataFilterOptions: function () {
+                return this.$store.getters['filteredData/selectedDataFilterOptions']
+            },
+
+            sexFilters: function (state): ChoroplethFiltersForType {
+                const available = (state.selectedDataType == DataType.ANC ?
+                    [
+                        {"id": "female", "name": "female"}
+                    ] :
+                    sexFilterOptions) as FilterOption[];
+                return this.buildViewFiltersForType(available, this.selectedChoroplethFilters.sex)
+            },
+
+            ageFilters: function (state): ChoroplethFiltersForType {
+                return this.buildViewFiltersForType(this.selectedDataFilterOptions.age,
+                    this.selectedChoroplethFilters.age);
+            },
+
+            surveyFilters: function (state): ChoroplethFiltersForType {
+                return this.buildViewFiltersForType(this.selectedDataFilterOptions.surveys,
+                    this.selectedChoroplethFilters.survey);
+            }
+        }),
+        methods: {
+            ...mapActions({
+                filterUpdated: 'filteredData/choroplethFilterUpdated',
+            }),
+            treeselectNormalizer(anyNode: any) {
+                const node = anyNode as FilterOption;
+                return {id: node.id, label: node.name};
+            },
+            buildViewFiltersForType(availableFilterOptions: FilterOption[],
+                                    selectedFilterOption?: FilterOption) {
+                return {
+                    available: availableFilterOptions,
+                    selected: selectedFilterOption ? selectedFilterOption.id : null,
+                    disabled: availableFilterOptions == undefined
+                }
+            },
+            updateFilter(filterType: FilterType, id: string, available: FilterOption[]) {
+                const newFilter = available.filter(o => o.id == id)[0];
+
+                this.filterUpdated([filterType, newFilter]);
+            },
+            updateSexFilter(id: string) {
+                this.updateFilter(FilterType.Sex, id, this.sexFilters.available);
+            },
+            updateAgeFilter(id: string) {
+                this.updateFilter(FilterType.Age, id, this.ageFilters.available);
+            },
+            updateSurveyFilter(id: string) {
+                this.updateFilter(FilterType.Survey, id, this.surveyFilters.available);
+            },
+            refreshSelectedChoroplethFilters(){
+                //if the selected data type has changed, we should update the choropleth filters if the dataset of that
+                //type does not include any of the selected filters as values. Set the filter to the first available value
+                const selectedChoroplethFilters = this.selectedChoroplethFilters;
+
+                const getNewFilter = function(filterName: string, available: FilterOption[]) {
```

Comment:
This could be it's own method no? Would make this long function easier to read. Also can it be called `getNewSelectedFilterOption`?